### ISSUE TITLE
firebase_web_app_config fails to use quota project override

### ISSUE THREAD

**Issue by serpro69** [hashicorp/terraform-provider-google#18717]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Terraform Version & Provider Version(s)

```
➜ tf -version
Terraform v1.9.1
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v5.37.0
+ provider registry.terraform.io/hashicorp/google-beta v5.37.0
+ provider registry.terraform.io/hashicorp/random v3.6.2

Your version of Terraform is out of date! The latest version
is 1.9.2. You can update by downloading from https://www.terraform.io/downloads.html

```

### Affected Resource(s)

google_firebase_web_app_config

### Terraform Configuration

```tf
resource "google_firebase_web_app" "main" {
  for_each     = toset(var.web_apps)
  provider     = google-beta.user_project_override
  project      = google_project.main.project_id
  display_name = each.value

  depends_on = [
    google_firebase_project.main
  ]
}

data "google_firebase_web_app_config" "main" {
  for_each   = toset(var.web_apps)
  provider   = google-beta.user_project_override
  web_app_id = google_firebase_web_app.main[each.value].app_id

  depends_on = [google_firebase_web_app.main]
}

```

### Expected Behavior

Should return firebase_web_app_config data when using provider with `user_project_override = true` config

### Actual Behavior

While the `google_firebase_web_app` resource is created fine, the `google_firebase_web_app_config` data returns the below error:

```
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Error when reading or editing dataSourceFirebaseWebAppConfig "1:554673022862:web:d3ae71e95c3dd6275ca6df"
│
│   with module.firebase_project.data.google_firebase_web_app_config.main["visit"],
│   on modules/firebase_project/webapp.tf line 12, in data "google_firebase_web_app_config" "main":
│   12: data "google_firebase_web_app_config" "main" {
│
│ googleapi: Error 403: Your application is authenticating by using local Application Default Credentials. The firebase.googleapis.com API requires a quota project, which is not set by default. To learn how to set your quota project, see
│ https://cloud.google.com/docs/authentication/adc-troubleshooting/user-creds .
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
│     "domain": "googleapis.com",
│     "metadata": {
│       "consumer": "projects/32555940559",
│       "service": "firebase.googleapis.com"
│     },
│     "reason": "SERVICE_DISABLED"
│   }
│ ]
│ , accessNotConfigured
╵

```

### Steps to reproduce

1. `terraform apply`

**Comment by ggtisc** [comment]
Hi @serpro69!

As you can see in [terraform registry](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/firebase_apple_app_config) this resource is on beta so for now you can wait until first version is released.

Also you could follow this [general guide](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_versions) for all beta resources until it will be available.

For now you could use this example as a guide:

```
provider "google" {
  user_project_override = true
  billing_project = "your-project-name"
  project = "your-project-name"
}

terraform {
  required_providers {
    google = {
      source  = "hashicorp/google-beta"
      version = "5.37.0"
    }
  }
}

resource "google_firebase_project" "fb_project_18717" {
  provider = google-beta
  project  = "your-project-name"
}

resource "google_firebase_web_app" "fb_wa_18717" {
  project = google_firebase_project.fb_project_18717.project
  provider = google-beta
  display_name = "fb-wa-18717"
}

data "google_firebase_web_app_config" "fb_wa_config_18717" {
  web_app_id = google_firebase_web_app.fb_wa_18717.app_id
}
```

**Comment by serpro69** [comment]
Hi @ggtisc ,
Well, that's the thing, I'm already using `google-beta` provider and setting `user_project_override = true`
Sorry if it wasn't clear from the code :( 
But you can see I'm setting this for each resource:
```
provider     = google-beta.user_project_override
```

Which has this config:
```
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "5.38.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "5.38.0"
      configuration_aliases = [
        google-beta,
        google-beta.user_project_override
      ]
    }
  }

  required_version = "~> 1.9"
}

provider "google" {
  alias                 = "user_project_override"
  user_project_override = true
  region                = "europe-north1"
  zone                  = "europe-north1-b"
  project               = local.project_name
}


```

And as I mentioned, the `google_firebase_web_app` resource works fine, which also requires google-beta provider with `user_project_override = true`. It's the `data "firebase_web_app_config"` that fails, even though they both use the same provider configuration.

**Comment by ggtisc** [comment]
Please try with the code I shared and see if it works

**Comment by serpro69** [comment]
I tried it, but your code doesn't work for me @ggtisc . For some reason it's trying to use billing account as project id when creating `google_firebase_project` resource:

```
google_firebase_project.fb_project_18717: Creating...

│ Error: Error checking if Firebase is already enabled: googleapi: Error 400: Project 'projects/ABC123-GHI456-MNO789' not found or deleted.

```

But in any case, I don't see any differences between your code and mine when it comes to using `data "google_firebase_web_app_config"`.

Also, as I've mentioned, I've tried all sorts of provider combinations for the `data "google_firebase_web_app_config"`, both `google-beta` with `user_project_overrides` and without it. Both fail with the same error, and only for this particular data source.

**Comment by ggtisc** [comment]
@serpro69 It looks like you are using a billing account instead of a billing project for the field `billing_project = "your-project-name"` please use again the same config and be sure that you are using the billing project and not the billing account.

**Comment by serpro69** [comment]
Hi @ggtisc ,
Thanks for your patience :) I tried now setting `billing_project` to the same value as `project` in the provider config, and that in fact did work!

Looking at the documentation for [`user_project_override`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#user_project_override), I can see it mentions that `billing_project` needs to be set for some cases:
> If true, most resources in the provider will explicitly supply their resource project, as described in their documentation. Otherwise, a billing_project value must be supplied.
I guess I should've paid more attention to this config parameter docs.

Thanks for the help! Closing this issue then.
